src/components/CheckBoxFilterContainer.ts

```diff
@@ -0,0 +1,183 @@
+                isChecked: this.props.defaultChecked
+            });
+        }
+
+        return null;
+    }
+
+    private applyFilter(isChecked: boolean) {
+        const { targetListView, targetNode } = this.state;
+        // To support multiple filters. We attach each checkbox-filter-widget's selected constraint
+        // On the listView's custom 'filter' object.
+        if (targetListView && targetListView._datasource) {
+            this.showLoader(targetNode);
+            const attribute = isChecked ? this.props.attribute : this.props.unCheckedAttribute;
+            const filterBy = isChecked ? this.props.filterBy : this.props.unCheckedFilterBy;
+            const constraint = isChecked ? this.props.constraint : this.props.unCheckedConstraint;
+            const attributeValue = isChecked ? this.props.attributeValue : this.props.unCheckedAttributeValue;
+
+            if (filterBy === "XPath") {
+                targetListView.filter[this.props.friendlyId] = constraint.indexOf(`[%CurrentObject%]'`) !== -1
+                    ? constraint.replace(`'[%CurrentObject%]'`, this.props.mxObject.getGuid())
+                    : constraint;
+            } else if (filterBy === "attribute") {
+                const ss = targetListView._datasource._pageObjs[0];
+                targetListView.filter[this.props.friendlyId] = (ss && ss.isEnum(attribute))
+                    ? `[${attribute}='${attributeValue.trim()}']`
+                    : (ss && ss.isBoolean(attribute))
+                        ? `[${attribute} = ${attributeValue.trim().toLowerCase()}()]`
+                        : `[contains(${attribute},'${attributeValue}')]`;
+            } else {
+                targetListView.filter[this.props.friendlyId] = "";
+            }
+
+            // Combine multiple-filter constraints into one and apply it to the listview
+            const finalConstraint = Object.keys(targetListView.filter)
+                .map(key => targetListView.filter[key])
+                .join("");
+            targetListView._datasource._constraints = targetListView.initialConstraint + finalConstraint;
+            // we have an issue with delayed update, that the grid sometimes initially displays unsorted content.
+            // We might have to make these calls synchronous instead of async.
+            targetListView.update(null, () => this.hideLoader(targetNode));
+        }
+    }
+
+    private connectToListView() {
+        const filterNode = findDOMNode(this).parentNode as HTMLElement;
+        const targetNode = Utils.findTargetNode(filterNode);
+        let targetListView: ListView | null = null;
+
+        if (targetNode) {
+            this.setState({ targetNode });
+            targetListView = dijitRegistry.byNode(targetNode);
+            if (targetListView) {
+                targetListView.filter = targetListView.filter || {};
+                if (targetListView.initialConstraint === undefined) {
```

Comment:
Do other widget also work this way? will it conflict?
Else you will copy the "original" that was modified by `Drop-down filter` widget?

Reply:
Verified and initialConstraint is not necessary. Listview maintains a staticConstraint for constraints coming from the server-side.
For the other widgets we'll need to add targetListView.filter[this.props.friendlyId]. so that they all don't conflict.